test/unit_tests/braket/circuits/test_from_ir.py

```diff
@@ -0,0 +1,86 @@
+import os
+import pickle
+
+from braket.circuits import Circuit
+from braket.devices import LocalSimulator
+
+devices = {
+    "default": LocalSimulator(),
+    "dm": LocalSimulator("braket_dm"),
+}
+
+
+def read_file(file_name):
+    f = open(file_name, "r")
+    file_text = f.read()
+    f.close()
+    return file_text
+
+
+def write_file(file_name, file_text):
+    f = open(file_name, "w")
+    f.write(file_text)
+    f.close()
+    return file_text
+
+
+def read_pickle(file):
+    return pickle.load(open(file, "rb"))
+
+
+def sort_key(d):
+    return {x: d[x] for x in sorted(d)} if d else None
+
+
+def test_from_ir():
+    print("Testing Circuit::from_ir()")
+
+    test_dir = "./test/unit_tests/braket/circuits/from"
+    input_dir = f"{test_dir}/input_ir"
+
+    check_dir = f"{test_dir}/check_ir.cache"
+    if not os.path.isdir(check_dir):
+        os.mkdir(check_dir, 0o755)
+    output_dir = f"{test_dir}/output_ir.cache"
+    if not os.path.isdir(output_dir):
+        os.mkdir(output_dir, 0o755)
+    result_dir = f"{test_dir}/result_ir.cache"
+    if not os.path.isdir(result_dir):
+        os.mkdir(result_dir, 0o755)
+
+    test_specs = read_file(f"{test_dir}/test_cases_ir.txt").strip("\n").split("\n")
+
+    for test_spec in test_specs:
```

Comment:
Instead of looping in a single test (which would make it hard to detect which case failed), consider how these tests can be executed as separate cases (`pytest.mark.parametrize` is one possibility).

Reply:
Yes, incorporated `pytest.mark.parametrize` to loop through the test instead of looping inside the test. Thanks for your advice @kshitijc 👍